pkg/istio/client.go

```diff
@@ -69,46 +63,38 @@ func NewVersionedIstioClient(cfg *rest.Config, config Config, recorder record.Ev
 	}, nil
 }
 
-type customClientErr struct {
-	Err        error
-	IsNotFound bool
-}
-
-func (c *Client) getVirtualService(ctx context.Context) (*istioclientapi.VirtualService, *customClientErr) {
+func (c *Client) GetVirtualService(ctx context.Context) (*istioclientapi.VirtualService, error) {
 	virtualService, err := c.NetworkingV1beta1().
 		VirtualServices(metav1.NamespaceDefault).
 		Get(ctx, c.config.VirtualServiceName, metav1.GetOptions{})
-	if client.IgnoreNotFound(err) != nil {
-		return nil, &customClientErr{
-			Err:        fmt.Errorf("failed to fetch virtual service %w", err),
-			IsNotFound: false,
-		}
-	}
-	if apierrors.IsNotFound(err) {
-		return nil, &customClientErr{
-			Err:        err,
-			IsNotFound: true,
-		}
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch virtual service %w", err)
 	}
 	return virtualService, nil
 }
 
-func (c *Client) createVirtualService(ctx context.Context, watcher *v1alpha1.Watcher,
+func (c *Client) CreateVirtualService(ctx context.Context, watcher *v1alpha1.Watcher,
 ) (*istioclientapi.VirtualService, error) {
 	if watcher == nil {
 		return &istioclientapi.VirtualService{}, nil
 	}
 
-	gateway, err := c.lookupGateway(ctx, watcher)
+	virtualSvc := &istioclientapi.VirtualService{}
+	virtualSvc.SetName(c.config.VirtualServiceName)
+	virtualSvc.SetNamespace(metav1.NamespaceDefault)
```

Comment:
Isn't the task about creating a distinct VirtualService For every watcher?
It looks like we're always creating the same one: name=c.config.VirtualServiceName, namespace="default"

Reply:
no, this will be handled in the followup issue, and it's not high priority as we discussed in daily. https://github.com/kyma-project/lifecycle-manager/issues/336